### PR TITLE
[ExecuTorch][WebGPU] batch_norm op tests

### PR DESCRIPTION
**Pull request by JCNTH**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* #20868
* #20996
* #20995
* __->__ #20994
* #20993
* #20992
* #20991
* #20990
* #20989
* #20988
* #20987
* #20986
* #20876
* #20875
* #20874
* #20873
* #20872
* #20871
* #20866
* #20865
* #20864
* #20863
* #20862
* #20861
* #20860
* #20859
* #20858
* #20857
* #20856
* #20855
* #20854
* #20852
* #20851
* #20850
* #20849
* #20848
* #20846
* #20845
* #20844
* #20843
* #20842

**Op-test suite for `aten._native_batch_norm_legit_no_training.default`** (stacked on the batch_norm op diff).

Adds `test/ops/test_batch_norm.py` (`BatchNorm2dModule` — `nn.BatchNorm2d.eval()` with deterministic running stats + affine) + a `@register_op_test("batch_norm")` suite in `test/op_tests/cases.py` (3 cases). Covers affine + non-affine (optional weight/bias) and an odd H*W; only the populated `out` ValueList entry is compared (out_index 0).
@exported-using-ghexport

Differential Revision: [D112417282](https://our.internmc.facebook.com/intern/diff/D112417282/)

Differential Revision: [D112417282](https://our.internmc.facebook.com/intern/diff/D112417282)

### CODE REVIEW

**Review by SS-JIA — APPROVED** [review]
Review automatically exported from Phabricator review in Meta.
